### PR TITLE
HW6: Routing

### PR DESCRIPTION
**Pull request by Myxjiew**
Added Angular routing;
Added blog page components and implemented some logic;
Added missed try-catch block to controllers;
ezgif-5-ff3cd2cd55

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="loading"></div>
+            </div>
```
Comment (MMaksymivUa) [inline src/app/app.component.html]: move to separate component

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+        <div class="row">
+            <div id="loader">
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
+                <div class="dot"></div>
```
Comment (MMaksymivUa) [inline src/app/app.component.html]: optional: you can use *ngFor

File: src/app/my-blog/components/blog-item/blog-item.component.scss [hunk 3]
```diff
@@ -0,0 +1,78 @@
+@import "../../../../styles/variables";
+
+.post {
+  margin: 0px 8px 8px 8px;
+  padding: 16px 2px 2px 2px;
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/blog-item/blog-item.component.scss:5]: margin: 0 8px 8px;
padding: 16px 2px 2px;

also works

File: src/app/my-blog/components/post-list/post-list.component.html [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+        <div class="dot"></div>
+        <div class="dot"></div>
+        <div class="dot"></div>
+        <div class="dot"></div>
+        <div class="dot"></div>
+        <div class="dot"></div>
+        <div class="dot"></div>
+        <div class="dot"></div>
+        <div class="loading"></div>
+      </div>
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/post-list/post-list.component.html:15]: reuse spinner component

File: src/app/my-blog/components/blog-item/blog-item.component.ts [hunk 5]
```diff
@@ -1,26 +1,27 @@
...
   public handleClick(post: IPost): void {
     post.toggle = !post.toggle;
     if (post.toggle) {
-      post.likeCounter++;
-    } else if (!post.toggle && post.likeCounter > 0) {
-      post.likeCounter--;
+      post.likes++;
+    } else if (!post.toggle && post.likes > 0) {
+      post.likes--;
     }
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/blog-item/blog-item.component.ts:25]: overcomplicated...
Why you need to check toggle? also:
```
if (post.toggle) {
      post.likes++;
    } else if (post.likes) {
      post.likes--;
```
Also, how it possible, if toggle true (if I correct understand - like button was clicked before), so post.likes can't be less than 0, so could be:
```
post.toggle ? post.likes++ : post.likes--;
```

File: src/app/my-blog/components/post-list/post-list.component.scss [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  & {
+    bottom: 0;
+    height: 175px;
+    left: 0;
+    margin: auto;
+    position: absolute;
+    right: 0;
+    top: 0;
+    width: 175px;
+  }
```
Comment (MMaksymivUa) [inline src/app/my-blog/components/post-list/post-list.component.scss:20]: ??? why it is here?

File: src/app/my-blog/post-page/post-page.component.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  public color = COLOR;
+
+  constructor(
+    private readonly blogService: BlogService,
+    private readonly route: ActivatedRoute,
+    private readonly router: Router
+  ) {}
+
+  public handleClick(post: IPost): void {
+    post.toggle = !post.toggle;
```
Comment (MMaksymivUa) [inline src/app/my-blog/post-page/post-page.component.ts:24]: also, the toggle is a bad name, of what ``toggle`` it is?

**Comment by MMaksymivUa** [comment]
Good job! Looks fine for me, but there are places in code, which do not meet code quality

File: src/app/my-blog/components/blog-item/blog-item.component.scss (hunk 3 above)
Comment (Myxjiew) [reply src/app/my-blog/components/blog-item/blog-item.component.scss:5]: Changed

File: src/app/my-blog/components/post-list/post-list.component.scss (hunk 6 above)
Comment (Myxjiew) [reply src/app/my-blog/components/post-list/post-list.component.scss:20]: Deleted

File: src/app/app.component.html (hunk 1 above)
Comment (Myxjiew) [reply src/app/app.component.html]: Moved

File: src/app/app.component.html (hunk 2 above)
Comment (Myxjiew) [reply src/app/app.component.html]: Changed

File: src/app/my-blog/components/post-list/post-list.component.html (hunk 4 above)
Comment (Myxjiew) [reply src/app/my-blog/components/post-list/post-list.component.html:15]: Done

File: src/app/my-blog/components/blog-item/blog-item.component.ts (hunk 5 above)
Comment (Myxjiew) [reply src/app/my-blog/components/blog-item/blog-item.component.ts:25]: Changed

**Pull request merged**